main.cpp

```diff
@@ -18,13 +18,34 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter_;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) {}
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+	const auto* CastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");

```

Comment:
What if `CastExpr` is a null pointer? How to protect the program from crashing in such cases?

Reply:
To protect the program from crash we need to add :
if(castExp == nullptr) {
return;
}